filters/colorinterp/ColorinterpFilter.cpp

```diff
@@ -0,0 +1,221 @@
+/******************************************************************************
+* Copyright (c) 2016, Howard Butler, hobu.inc@gmail.com
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "ColorinterpFilter.hpp"
+
+#include <pdal/PointView.hpp>
+#include <pdal/GDALUtils.hpp>
+#include <pdal/pdal_macros.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+
+#include <gdal.h>
+#include <cpl_vsi.h>
+#include <ogr_spatialref.h>
+
+#include <array>
+
+#include "color-ramps.hpp"
+
+namespace pdal
+{
+
+static std::vector<std::string> ramps = {"awesome_green", "black_orange", "blue_hue", "blue_red", "heat_map", "pestel_shades", "blue_orange"};
+
+static PluginInfo const s_info = PluginInfo(
+    "filters.colorinterp",
+    "Assigns RGB colors based on a dimension and a ramp",
+    "http://pdal.io/stages/filters.colorinterp.html" );
+
+CREATE_STATIC_PLUGIN(1, 0, ColorinterpFilter, Filter, s_info)
+
+std::string ColorinterpFilter::getName() const { return s_info.name; }
+
+
+#define GETRAMP(name) \
```

Comment:
I wrote a macro because the variable name for the byte arrays is the ramp name, and I'm lazy.

The VSIFILE* that is created just ends up hanging out there, but the bytes it points at of course are our byte arrays in color-ramps.hpp.